service/src/main/java/com/derpgroup/derpwizard/resource/AlexaResource.java

```diff
@@ -74,35 +85,47 @@ public AlexaResource(MainConfig config, Environment env) {
    * Generates a welcome message.
    *
    * @return The message, never null
-   * @throws IOException 
    */
   @POST
-  public SpeechletResponseEnvelope doAlexaRequest(@NotNull @Valid SpeechletRequestEnvelope request) throws IOException{
-    if (request.getRequest() == null) {
-      throw new RuntimeException("Missing request body."); //TODO: create AlexaException
+  public SpeechletResponseEnvelope doAlexaRequest(@NotNull @Valid SpeechletRequestEnvelope request){
+
+    ObjectMapper mapper = new ObjectMapper();
+    CommonMetadata metadata = null;
+    try{
+      if (request.getRequest() == null) {
+        throw new DerpwizardException(DerpwizardExceptionReasons.MISSING_INFO.getSsml(),"Missing request body."); //TODO: create AlexaException
+      }
+      
+      Map<String, Object> sessionAttributes = request.getSession().getAttributes();
+      metadata = mapper.convertValue(sessionAttributes, new TypeReference<CommonMetadata>(){});
+      SsmlDocumentBuilder builder = new SsmlDocumentBuilder(UNSUPPORTED_SSML_TAGS);
+      VoiceInput voiceInput = VoiceMessageFactory.buildInputMessage(request.getRequest(), InterfaceType.ALEXA);
+      manager.handleRequest(voiceInput, builder);
+  
+      SpeechletResponseEnvelope responseEnvelope = new SpeechletResponseEnvelope();
+      responseEnvelope.setSessionAttributes(request.getSession().getAttributes());
+  
+      // Create a VoiceOutput object with the SSML content generated by the manater
+      if ((voiceInput.getMessageType() != MessageType.END_OF_CONVERSATION) && (voiceInput.getMessageType() != MessageType.STOP)) {
+        SimpleCard card = new SimpleCard();
+        card.setContent(builder.getRawText());
+        card.setTitle("Alexa + DERP Wizard");
+  
+        @SuppressWarnings("unchecked")
+        VoiceOutput<SpeechletResponse> voiceOutput = (VoiceOutput<SpeechletResponse>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
+        SpeechletResponse speechletResponse = (SpeechletResponse) voiceOutput.getImplInstance();
+        speechletResponse.setCard(card);
+  
+        responseEnvelope.setResponse(speechletResponse);
+      }
+  
+      return responseEnvelope;
+    }catch(DerpwizardException e){
+      LOG.debug(e.getMessage());
+      return new DerpwizardExceptionAlexaWrapper(e, "1.0",mapper.convertValue(metadata, new TypeReference<Map<String,Object>>(){}));
+    }catch(Throwable t){
+      LOG.debug(t.getMessage());
+      return new DerpwizardExceptionAlexaWrapper(new DerpwizardException(t.getMessage()),"1.0", mapper.convertValue(metadata, new TypeReference<Map<String,Object>>(){}));
```

Comment:
If we receive an Amazon Stop or Cancel intent, and we generate an exception while handling that, it'll bounce back up to this level and return a real response (from the DerpwizardExceptionAlexaWrapper). The Echo doesn't want a response to those amazon intents, and the Echo will blow up saying "There was a problem with the requested skill".

We have control at this level, and we know what the intent was. Inside these catch statements we can choose if we want to return an empty response, or the DerpwizardExceptionAlexaWrapper.

We can't return a response to the user on a close/cancel/abort intent anyways, so we should return an empty response to them instead.